src/main/java/org/breedinginsight/daos/ProgramUserDAO.java

```diff
@@ -43,6 +43,14 @@ public ProgramUserDAO(Configuration config, DSLContext dsl) {
         this.dsl = dsl;
     }
 
+    public boolean existsAndActive(UUID programId, UUID userId) {
+        return dsl.selectCount().from(PROGRAM_USER_ROLE)
+                .where(PROGRAM_USER_ROLE.PROGRAM_ID.eq(programId)
+                        .and(PROGRAM_USER_ROLE.USER_ID.eq(userId))
+                        .and(PROGRAM_USER_ROLE.ACTIVE.eq(true)))
+                .fetchOne(0, Integer.class) == 1;
```

Comment:
This will return false if the user has more than 1 program role. In the code we have only one role, but our database accepts more than 1 role. Maybe do `> 1` to future proof it?

Reply:
Nevermind, I missed the `fetchOne`

Reply:
One other thing on this to be aware of is that currently all the roles of a user are set to active/deactivate in a program so this will work fine. If things ever changes to where different roles could have different active status, this would need to change.

Reply:
Do you mean like a null in the field? I think we are safe from this because our `active` fields in the db are set to `NOT NULL BOOLEAN` so will always be true false.

Reply:
Talked with @ctucker3 and we resolved this.